### PR TITLE
chore: Update sphinx dependencies to use 'dev' branch

### PR DESCRIPTION
**Pull request by huitseeker**
- Upgraded the dependency versions of `sphinx-derive`, `sphinx-sdk`, `sphinx-zkvm`, `sphinx-helper`, and `sphinx-prover` across multiple programs and modules from release `v1.0.0` to development branch `dev`.

We need this for the "check-downstream" jobs of sphinx to work in any fashion.

### CODE REVIEW

**Review by adr1anh — APPROVED** [review]

**Review by samuelburnham — APPROVED** [review]
Stamping to unblock, but note to check the cycle counts once the workflow is fixed and also refactor the release workflow to not merge the release branch into dev.

**Pull request merged**
